app/api/scorecard.py

```diff
@@ -0,0 +1,140 @@
+    # Create the scorecard.
+    scoreCardInput.code = format_code(scoreCardInput.name)
+    scorecard = schemas.ScoreCardCreate(
+        name=scoreCardInput.name,
+        code=scoreCardInput.code,
+        description=scoreCardInput.description
+    )
+    if (scoreCardCrud.getByScoreCardCode(scoreCardInput.code)):
+        raise HTTPResponseCustomized(
+            status_code=400, detail="Scorecard already found")
+    try:
+        obj = scoreCardCrud.create(scorecard)
+        scorecardID = obj.id
+    except Exception as e:
+        raise HTTPResponseCustomized(status_code=422, detail=f"Error: {e}")
+
+    # if the scorecardids are passed duplicated
+    servicesIDsSet = list(set(scoreCardInput.services))
+    # i need to check if the serviceid is found or not
+    service = serviceCrud.getByServiceIds(servicesIDsSet)
+    serviceListToBeCreated = []
+    for serviceid in servicesIDsSet:
+        if (len(service) == len(servicesIDsSet)):
+            serviceScorecard = schemas.MicroserviceScoreCardCreate(
+                scoreCardId=scorecardID,
+                microserviceId=serviceid
+            )
+            serviceListToBeCreated.append(serviceScorecard)
+        else:
+            scoreCardCrud.delete(scorecardID)
+            raise HTTPResponseCustomized(
+                status_code=400, detail="Service not found")
+
+    objects = []
+    for metric in scoreCardInput.metrics:
+        metric_data = metricCrud.getById(metric.id)  # Get the metric data
+        if (metric_data == None):
+            scoreCardCrud.delete(scorecardID)
+            raise HTTPResponseCustomized(
+                status_code=404, detail="Metric is not found")
+        metricCreate = schemas.MetricTypeScorecard(
+            id=metric.id,
+            weight=metric.weight,
+            desiredValue=metric.desiredValue,
+            criteria=metric.criteria,
+            type=metric_data.type  # Add the type directly from retrieved metric
+        )
+        objects.append(metricCreate)
+    try:
+        if (objects):
+            check_metric(objects)
+    except Exception as e:
+        scoreCardCrud.delete(scorecardID)
+        check_metric(objects)  # Return the error message
+    for metric in scoreCardInput.metrics:
+        if (metricCrud.getById(metric.id)):
+            metric.desiredValue = stringify_value(metric.desiredValue)
+            scorecardmetric = schemas.ScoreCardMetricsCreate(
+                scoreCardId=scorecardID,
+                metricId=metric.id,
+                criteria=metric.criteria,
+                weight=metric.weight,
+                desiredValue=metric.desiredValue
+            )
+        else:
+            scoreCardCrud.delete(scorecardID)
+            raise HTTPResponseCustomized(
+                status_code=404, detail="Metric not found")
+        try:
+            scorecardMetricCrud.create(scorecardmetric)
+        except Exception as e:
+            scoreCardCrud.delete(scorecardID)
```

Comment:
you are inside a loop, you need to think about this carefully.
what if the first iteration passed successfully but the second one failed in the create method, should we delete the scorecard object ? it's already linked with the scorecard_metric from the first iteration!